twisted/spread/jelly.py

```diff
@@ -889,18 +915,9 @@ def _unjelly_instance(self, rest):
             category=DeprecationWarning, filename="", lineno=0)
 
         clz = self.unjelly(rest[0])
-        if type(clz) is not _OldStyleClass:
-            raise InsecureJelly("Instance found with non-class class.")
-        if hasattr(clz, "__setstate__"):
-            inst = _newInstance(clz, {})
-            state = self.unjelly(rest[1])
-            inst.__setstate__(state)
-        else:
-            state = self.unjelly(rest[1])
-            inst = _newInstance(clz, state)
-        if hasattr(clz, 'postUnjelly'):
-            self.postCallbacks.append(inst.postUnjelly)
-        return inst
+        if not _PY3 and type(clz) is not _OldStyleClass:
+            raise InsecureJelly("Legacy 'instance' found with new-style class")
```

Comment:
I don't see any tests for this change ... if we are testing just for InsecureJelly I think that we should also assert the InsecureJelly's message